presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/AddLocalExchanges.java

```diff
@@ -245,23 +238,16 @@ public PlanWithProperties visitAggregation(AggregationNode node, StreamPreferred
             StreamPreferredProperties requiredProperties;
             StreamPreferredProperties preferredChildProperties;
 
-            if (node.getStep() == Step.FINAL || node.getStep() == Step.SINGLE) {
-                // final aggregation requires that all data be partitioned
-                HashSet<Symbol> partitioningRequirement = new HashSet<>(node.getGroupingSets().get(0));
-                for (int i = 1; i < node.getGroupingSets().size(); i++) {
-                    partitioningRequirement.retainAll(node.getGroupingSets().get(i));
-                }
-
-                requiredProperties = parentPreferences.withDefaultParallelism(session).withPartitioning(partitioningRequirement);
-                preferredChildProperties = parentPreferences.withDefaultParallelism(session)
-                        .withPartitioning(partitioningRequirement);
-            }
-            else {
-                requiredProperties = parentPreferences.withoutPreference().withDefaultParallelism(session);
-                preferredChildProperties = parentPreferences.withDefaultParallelism(session)
-                        .constrainTo(node.getSource().getOutputSymbols());
+            // final aggregation requires that all data be partitioned
```

Comment:
lets add a check here that the node step is SINGLE. this way we can detect errors where people accidentally move around optimizers and break this assumption